### ISSUE TITLE
file.manage: source_hash not working with s3:// (2014.7.0rc1)

### ISSUE THREAD

**Issue by hubez** [saltstack/salt#15209]
file.managed source_hash does not work properly if I use s3 (ex: s3://mybucket/myfile.sha1).

This is the output I get if I run the state module:

```
    Function: file.managed
      Result: False
     Comment: Source hash file s3://mybucket/myfile.sha1 contains an invalid hash format, it must be in the format <hash type>=<hash>
     Started: 21:31:00.026150
     Duration: 0.941 ms
```

This is the .sls file:
```
test:
  user:
    - present
  require:
    - group: testgroup

testgroup:
  group:
    - present

/home/test/s3file:
  file.managed:
    - user: test
    - group: testgroup
    - mode: 644
    - source: s3://mybucket/myfile
    - source_hash: s3://mybucket/myfile.sha1
```

```
$ cat myfile.sha1
sha1=ac3aa3b3b0afa69b1aa11f19a9005b9ffd0633aa
```

I believe this is an issue with using s3:// (and not a permission issue) since source_hash works if I just specify the hash:
```
/home/test/s3file:
  file.managed:
    - user: test
    - group: testgroup
    - mode: 644
    - source: s3://mybucket/myfile
    - source_hash: sha1=ac3aa3b3b0afa69b1aa11f19a9005b9ffd0633aa
``` 

Additionally, I believe myfile.sha1 is properly formatted because if I use salt:// it works:
```
/home/test/s3file:
  file.managed:
    - user: test
    - group: testgroup
    - mode: 644
    - source: s3://mybucket/myfile
    - source_hash: salt://myfile.sha1
```

**Comment by TronPaul** [comment]
I think I fixed this with #19640. @hubez if you try this, let me know if it doesn't work for you.

**Comment by TronPaul** [comment]
Is this going to get into [v2015.8.0rc1](https://github.com/saltstack/salt/tree/v2015.8.0rc1)? I'd love to be able to stop using a custom branch on a fork to avoid using develop.
